tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml

```diff
@@ -33,90 +33,110 @@ jobs:
     displayName: 'Build onnxruntime'
-  
+    
   - bash: |-
       echo "##vso[task.setvariable variable=onnxruntimeBuildSucceeded]true"
     displayName: 'Set Onnxruntime Build Succeeded'
-
-  - script: |-
-      cd ./build/RelWithDebInfo &&\
-      ../../tools/ci_build/github/pai/pai_test_launcher.sh
+  
+  - task: CmdLine@2
+    inputs:
+      script: |-
+        cd ./build/RelWithDebInfo &&\
+        ../../tools/ci_build/github/pai/pai_test_launcher.sh
     displayName: 'Run onnxruntime unit tests'
-
-  - script: |-
-      cd ./build/RelWithDebInfo
-      export PYTHONPATH=$PWD
-      python -m onnxruntime.training.ortmodule.torch_cpp_extensions.install
+    
+  - task: CmdLine@2
+    inputs:
+      script: |-
+        cd ./build/RelWithDebInfo
+        export PYTHONPATH=$PWD
+        python -m onnxruntime.training.ortmodule.torch_cpp_extensions.install
     displayName: 'Compile torch extensions into build directory'
     condition: and(succeededOrFailed(), eq(variables.onnxruntimeBuildSucceeded, 'true')) # ensure all tests are run when the build successed
   
-  - script: |-
-      cd ./build/RelWithDebInfo
-      export PYTHONPATH=$PWD
-      export ORTMODULE_ONNX_OPSET_VERSION=12
-      python \
-        /stage/huggingface-transformers/examples/pytorch/language-modeling/run_mlm.py \
-        --model_name_or_path bert-large-uncased \
-        --dataset_name wikitext \
-        --dataset_config_name wikitext-2-raw-v1 \
-        --do_train \
-        --max_steps 260 \
-        --logging_steps 20 \
-        --output_dir ./test-mlm-bbu \
-        --overwrite_output_dir \
-        --per_device_train_batch_size 8 \
-        --fp16 \
-        --dataloader_num_workers 1 \
-        --ort \
-        --skip_memory_metrics
-      python ../../orttraining/tools/ci_test/compare_huggingface.py \
-        ci-pipeline-actual.json \
-        ../../orttraining/tools/ci_test/results/ci-mi100.huggingface.bert-large-rocm4.3.1.json
+  - task: CmdLine@2
+    inputs:
+      script: |-
+        cd ./build/RelWithDebInfo
+        export PYTHONPATH=$PWD
+        export ORTMODULE_ONNX_OPSET_VERSION=12
+        python \
+          /stage/huggingface-transformers/examples/pytorch/language-modeling/run_mlm.py \
+          --model_name_or_path bert-large-uncased \
+          --dataset_name wikitext \
+          --dataset_config_name wikitext-2-raw-v1 \
+          --do_train \
+          --max_steps 260 \
+          --logging_steps 20 \
+          --output_dir ./test-mlm-bbu \
+          --overwrite_output_dir \
+          --per_device_train_batch_size 8 \
+          --fp16 \
+          --dataloader_num_workers 1 \
+          --ort \
+          --skip_memory_metrics
+        python ../../orttraining/tools/ci_test/compare_huggingface.py \
+          ci-pipeline-actual.json \
+          ../../orttraining/tools/ci_test/results/ci-mi100.huggingface.bert-large-rocm4.3.1.json
     displayName: 'Run Python Hugging-Face BERT-L test'
+    retryCountOnTaskFailure: 2
```

Comment:
>2 [](http://example.com/codeflow?start=29&length=1)

should we reduce it to 1 ?

Reply:
Yes, may be 1 time is enough, avoid repeated failure when the pr does have error